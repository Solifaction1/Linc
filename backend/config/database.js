const path = require('path');

const connection = process.env.NODE_ENV !== 'production' ?
  {
    client: 'sqlite',
    connection: {
      filename: path.join(__dirname, '..', '.tmp/data.db'),
    },
    useNullAsDefault: true,
  } : {
    client: "postgres",
    acquireConnectionTimeout: 120000,
    connection: {
      // host: `/cloudsql/${process.env.INSTANCE_CONNECTION_NAME}`, // for a PostgreSQL database
      host: process.env.DATABASE_HOST, // for a PostgreSQL database
      database: process.env.DATABASE_NAME,
      user: process.env.DATABASE_USERNAME,
      password: process.env.DATABASE_PASSWORD,
      ssl: false,

    },
    useNullAsDefault: true,
    debug: false,
  };

module.exports = ({ }) => ({
  connection: connection,
});
