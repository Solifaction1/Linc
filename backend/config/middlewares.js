module.exports = [
  'strapi::errors',
  {
    name: 'strapi::cors',
    config: {
      enabled: true,
      headers: ['*'],
      origin: ['*']
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            "cdn.jsdelivr.net",
            "strapi.io",
            "s3.amazonaws.com",
            "editor.unlayer.com",
            process.env.CDN_URL,
          ],
          "script-src": ["'self'", "editor.unlayer.com"],
          "frame-src": ["'self'", "editor.unlayer.com"],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            process.env.CDN_URL,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
];
