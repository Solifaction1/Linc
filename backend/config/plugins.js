module.exports = ({ env }) => ({
  "strapi-plugin-populate-deep": {
    config: {
      defaultDepth: 5, // Default is 5
    },
  },
  "duplicate-button": true,
  email: {
    config: {
      provider: "mailgun",
      providerOptions: {
        key: env("MAILGUN_API_KEY"), // Required
        domain: env("MAILGUN_DOMAIN"), // Required
        url: "https://api.eu.mailgun.net", //Optional. If domain region is Europe use 'https://api.eu.mailgun.net'
      },
      settings: {
        defaultFrom: "<mg@linc.legal>",
        defaultReplyTo: "<mg@linc.legal>",
      },
    },
  },
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
  "content-versioning": {
    enabled: false,
  },
  slugify: {
    enabled: false,
    config: {
      contentTypes: {
        blog: {
          field: "slug",
          references: "title",
        },
      },
    },
  },
  "preview-button": {
    enabled: true,
    resolve: "./src/plugins/preview-button", // path to plugin folder
  },
  "email-designer": {
    enabled: true,

    // ⬇︎ Add the config property
    config: {
      editor: {
        // optional - if you have a premium unlayer account
        // projectId: [UNLAYER_PROJECT_ID],

        tools: {
          heading: {
            properties: {
              text: {
                value: "This is the new default text!",
              },
            },
          },
        },
        options: {
          features: {
            colorPicker: {
              presets: ["#D9E3F0", "#F47373", "#697689", "#37D67A"],
            },
          },
          fonts: {
            showDefaultFonts: false,
            /*
             * If you want use a custom font you need a premium unlayer account and pass a projectId number :-(
             */
            customFonts: [
              {
                label: "Anton",
                value: "'Anton', sans-serif",
                url: "https://fonts.googleapis.com/css?family=Anton",
              },
              {
                label: "Lato",
                value: "'Lato', Tahoma, Verdana, sans-serif",
                url: "https://fonts.googleapis.com/css?family=Lato",
              },
              // ...
            ],
          },
          mergeTags: [
            {
              name: "Naam",
              value: "{{= USER.Naam }}",
              sample: "Naam",
            },
            {
              name: "Voornaam",
              value: "{{= USER.Voornaam }}",
              sample: "Voornaam",
            },
            {
              name: "Email",
              value: "{{= USER.Email }}",
              sample: "john@doe.com",
            },
            {
              name: "Telefoonnummer",
              value: "{{= USER.Telefoonnummer }}",
              sample: "0123456789",
            },
            {
              name: "select",
              value: "{{= USER.select }}",
              sample: "select",
            },
            {
              name: "textArea",
              value: "{{= USER.textArea }}",
              sample: "textArea",
            },
            {
              name: "Form ID",
              value: "{{= USER.formId }}",
              sample: "0",
            },
          ],
        },
        appearance: {
          theme: "dark",
          panels: {
            tools: {
              dock: "left",
            },
          },
        },
      },
    },
  },
  seo: {
    enabled: true,
  },
  ckeditor: {
    enabled: true,
    config: {
      plugin: {
        // disable data-theme tag setting //
        // setAttribute:false,
        // disable strapi theme, will use default ckeditor theme //
        // strapiTheme:false,
        // styles applied to editor container (global scope) //
        // styles:`
        // :root{
        //   --ck-color-focus-border:red;
        //   --ck-color-text:red;
        // }
        // `
        styles: `
        .ck-editor__main p,
        .ck.ck-list__item .ck-button .ck-button__label {
          line-height: normal;
        }`,
      },
      editor: {
        // editor default config

        // https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html
        // if you need markdown support and output set: removePlugins: [''],
        // default is
        removePlugins: ["Markdown"],

        // https://ckeditor.com/docs/ckeditor5/latest/features/toolbar/toolbar.html
        toolbar: {
          items: [
            "paragraph",
            "heading1",
            "heading2",
            "heading3",
            "|",
            "bold",
            "italic",
            "underline",
            "fontSize",
            "removeFormat",
            "|",
            "bulletedList",
            "numberedList",
            "alignment",
            "|",
            "StrapiMediaLib",
            "link",
            "horizontalLine",
            "|",
            "sourceEditing",
            "fullScreen",
            "undo",
            "redo",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/font.html
        fontSize: {
          options: [
            9,
            11,
            13,
            {
              title: "default",
              model: "18px",
            },
            17,
            19,
            21,
            27,
            35,
          ],
          supportAllValues: false,
        },
        fontFamily: {
          options: [
            "default",
            "Arial, Helvetica Neue, Helvetica, Source Sans Pro, sans-serif",
            "Courier New, Courier, monospace",
            "Georgia, serif",
            "Lucida Sans Unicode, Lucida Grande, sans-serif",
            "Tahoma, Geneva, sans-serif",
            "Times New Roman, Times, serif",
            "Trebuchet MS, Helvetica, sans-serif",
            "Verdana, Geneva, sans-serif",
            "Roboto, Roboto Black, Roboto Medium, Roboto Light, sans-serif",
          ],
          supportAllValues: true,
        },
        fontColor: {
          columns: 5,
          documentColors: 10,
        },
        fontBackgroundColor: {
          columns: 5,
          documentColors: 10,
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
        // default language: 'en',
        // https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html
        image: {
          resizeUnit: "%",
          resizeOptions: [
            {
              name: "resizeImage:original",
              value: null,
              icon: "original",
            },
            {
              name: "resizeImage:25",
              value: "25",
              icon: "small",
            },
            {
              name: "resizeImage:50",
              value: "50",
              icon: "medium",
            },
            {
              name: "resizeImage:75",
              value: "75",
              icon: "large",
            },
          ],
          toolbar: [
            "toggleImageCaption",
            "imageTextAlternative",
            "imageStyle:inline",
            "imageStyle:block",
            "imageStyle:side",
            "linkImage",
            "resizeImage:25",
            "resizeImage:50",
            "resizeImage:75",
            "resizeImage:original",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/table.html
        table: {
          contentToolbar: [
            "tableColumn",
            "tableRow",
            "mergeTableCells",
            "tableCellProperties",
            "tableProperties",
            "toggleTableCaption",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/headings.html
        heading: {
          options: [
            {
              model: "paragraph",
              title: "Paragraph",
              class: "ck-heading_paragraph",
            },
            {
              model: "heading1",
              view: "h1",
              title: "Heading 1",
              class: "ck-heading_heading1",
            },
            {
              model: "heading2",
              view: "h2",
              title: "Heading 2",
              class: "ck-heading_heading2",
            },
            {
              model: "heading3",
              view: "h3",
              title: "Heading 3",
              class: "ck-heading_heading3",
            },
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html
        htmlSupport: {
          allow: [
            {
              name: "img",
              attributes: {
                sizes: true,
                loading: true,
              },
            },
          ],
        },
      },
    },
  },
});
