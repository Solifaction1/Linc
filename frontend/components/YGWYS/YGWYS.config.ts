type YGWYSData = {
	text: string
	className?: string | undefined
	inline?: boolean | undefined
}

const defaultProps: YGWYSData = {
	text: '<p>Do you get what you see?</p>',
	className: 'description',
	inline: false,
}

export type { YGWYSData }
export { defaultProps }
