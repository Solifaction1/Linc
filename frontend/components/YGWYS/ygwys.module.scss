@import 'styles/vars';
@import 'styles/utilities';

.ygwys {
	display: block;

	&.inline {
		@extend %page-container;
		display: block;
		grid-template-columns: 1fr;

		@media (min-width: $break-md) {
			h1,
			h2,
			h3 {
				width: 75rem;
			}

			ul {
				width: 68rem;
			}

			h1,
			h2,
			h3,
			p,
			ul,
			ol,
			img {
				max-width: 75rem;
				margin: 0 auto;
				margin-bottom: 1.6rem;
			}

			p {
				margin-right: 0;
				margin-left: 0;
			}
		}

		h1,
		h2 {
			font-weight: $font-weight-semi-bold;
		}
	}

	a {
		color: $color-gray;
		text-decoration: underline;
	}

	img {
		max-width: 100%;
	}

	figure {
		&[class*='image'] {
			margin: 0 auto;
		}

		&[class*='image-style-side'] {
			margin: unset;
			margin-left: auto;
		}

		img {
			width: 100%;
		}
	}

	strong {
		font-weight: $font-weight-bold;
	}

	table {
		@extend %body-copy-3;
		width: 100%;
		max-width: 100%;
		margin: 2rem 0;

		td {
			padding: 1rem;
		}
	}

	i {
		font-style: italic;
	}

	ul {
		@extend %list-custom-bullet;
		height: 100%;
		list-style-type: disc;

		li {
			@extend %body-copy-2;
			padding-left: calc(#{$spacing-base} * 1);
			font-weight: $font-weight-light;
		}
	}

	ol {
		max-width: 100%;
		margin-left: 2rem;
		counter-reset: item;

		li {
			@extend %body-copy-2;

			max-width: 100%;
			padding-left: calc(#{$spacing-base} * 2);

			list-style-position: outside;

			font-weight: $font-weight-light;

			counter-increment: item;

			&::marker {
				content: counter(item) '.';
			}

			ol {
				padding-left: calc(#{$spacing-base} * 2);

				& > li {
					margin-top: 1.8rem;
				}

				li {
					@extend %body-copy-2;

					max-width: 100%;
					padding-left: calc(#{$spacing-base} * 2);

					list-style-position: outside;

					font-weight: $font-weight-light;

					&::marker {
						content: counters(item, '.') ' ';
					}
				}
			}
		}
	}

	h1,
	h2,
	h3 {
		font-weight: $font-weight-medium;

		b {
			font-weight: $font-weight-semi-bold;
		}

		u {
			background-position-x: 4px, 0, 100%;
			background-position-y: calc(100% - 0px);
			background-size: calc(100% - 8px) 8px, auto 8px, auto 8px;
			@media (max-width: $break-md) {
				background-position-x: 3px, 0, 100%;
				background-size: calc(100% - 6px) 6px, auto 6px, auto 6px;
			}
		}
	}

	h1 {
		@extend %h1;
		margin-bottom: calc(#{$spacing-base} * 4);

		mark {
			background-position-y: calc(100% - 18px);
			@media (max-width: $break-md) {
				background-position-y: calc(100% - 14px);
			}
		}
	}

	h2 {
		@extend %h2;
		margin-bottom: calc(#{$spacing-base} * 4);

		mark {
			background-position-y: calc(100% - 14px);
			@media (max-width: $break-md) {
				background-position-y: calc(100% - 11px);
			}
		}
	}

	h3 {
		@extend %h3;
		margin-bottom: calc(#{$spacing-base} * 3);

		mark {
			background-position-y: calc(100% - 10px);
			background-size: calc(100% - 6px) 6px, auto 6px, auto 6px;
			@media (max-width: $break-md) {
				background-position-x: 2px, 0, 100%;
				background-position-y: calc(100% - 8px);
				background-size: calc(100% - 4px) 4px, auto 4px, auto 4px;
			}
		}

		u,
		a {
			background-position-x: 3px, 0, 100%;
			background-size: calc(100% - 6px) 6px, auto 6px, auto 6px;
			@media (max-width: $break-md) {
				background-position-x: 2px, 0, 100%;
				background-size: calc(100% - 4px) 4px, auto 4px, auto 4px;
			}
		}
	}

	p {
		@extend %body-copy-1;
	}

	blockquote {
		position: relative;

		width: auto;
		max-width: 100%;
		margin: calc(#{$spacing-base} * 8);
		margin-top: calc(#{$spacing-base} * 8 + 6rem);
		padding: calc(#{$spacing-base} * 9) calc(#{$spacing-base} * 22);

		p,
		u,
		a {
			@extend %body-copy-1;
			grid-area: p;
			margin-top: unset;
			color: white;
		}
	}
}
