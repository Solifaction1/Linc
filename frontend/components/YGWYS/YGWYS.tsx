import React from 'react'
import sanitizeHtml from 'sanitize-html'
import { YGWYSData } from './YGWYS.config'
import styles from './ygwys.module.scss'

const YGWYS = (props: YGWYSData) => {
	return (
		<div
			className={`${styles.ygwys} 
			${props.className} ${props.inline && styles.inline}`}
			dangerouslySetInnerHTML={{
				__html: sanitizeHtml(props.text, {
					allowedTags: [
						'img',
						'video',
						'address',
						'article',
						'aside',
						'footer',
						'header',
						'h1',
						'h2',
						'h3',
						'h4',
						'h5',
						'h6',
						'hgroup',
						'main',
						'nav',
						'section',
						'blockquote',
						'dd',
						'div',
						'dl',
						'dt',
						'figcaption',
						'figure',
						'hr',
						'li',
						'main',
						'ol',
						'p',
						'pre',
						'ul',
						'a',
						'abbr',
						'b',
						'bdi',
						'bdo',
						'br',
						'cite',
						'code',
						'data',
						'dfn',
						'em',
						'i',
						'kbd',
						'mark',
						'q',
						'rb',
						'rp',
						'rt',
						'rtc',
						'ruby',
						's',
						'samp',
						'small',
						'span',
						'strong',
						'sub',
						'sup',
						'time',
						'u',
						'var',
						'wbr',
						'caption',
						'col',
						'colgroup',
						'table',
						'tbody',
						'td',
						'tfoot',
						'th',
						'thead',
						'tr',
					],
					disallowedTagsMode: 'discard',
					allowedAttributes: {
						a: ['href', 'name', 'target'],
						figure: ['class', 'style', 'className'],
						// We don't currently allow img itself by default, but
						// these attributes would make sense if we did.
						img: [
							'src',
							'srcset',
							'alt',
							'title',
							'width',
							'height',
							'loading',
							'style',
							'sizes',
							'class',
						],
						p: ['style', 'class'],
						ol: ['style'],
						ul: ['style'],
						span: ['style', 'class'],
					},
				}),
			}}
		/>
	)
}

export default YGWYS
