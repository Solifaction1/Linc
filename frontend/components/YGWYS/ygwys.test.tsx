import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import YGWYS from './YGWYS'
import { defaultProps } from './YGWYS.config'

describe('ygwys', () => {
	it('renders a YGWYS component', () => {
		const { container } = render(<YGWYS {...defaultProps} />)
		expect(container).toMatchSnapshot()
		expect(container.getElementsByClassName('ygwys')[0]).toBeVisible()
	})
})
