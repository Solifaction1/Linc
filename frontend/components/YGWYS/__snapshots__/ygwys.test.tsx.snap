// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ygwys renders a YGWYS component 1`] = `
<div>
  <div
    class="ygwys 
			description false"
  >
    <p>
      Do you get what you see?
    </p>
  </div>
</div>
`;
